packages/loader/container-loader/src/container.ts

```diff
@@ -1493,12 +1523,16 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         deltaManager.inboundSignal.pause();
 
-        deltaManager.on("connect", (details: IConnectionDetails, opsBehind?: number) => {
-            // Back-compat for new client and old server.
-            this._audience.clear();
+        deltaManager.on("connect", (details: IConnectionDetails, _opsBehind?: number) => {
+            this._initialClients = details.initialClients ?? [];
 
-            for (const priorClient of details.initialClients ?? []) {
-                this._audience.addMember(priorClient.clientId, priorClient.client);
+            if (this._protocolHandler !== undefined) {
```

Comment:
I think it will not be Ok to drop initial clients on the floor if we do not have connection. This will not impact 1st party scenarios (that all delay connection until after boot), but default mode runs connection in parallel with rest of boot, and this will miss initial audience population.

Reply:
They aren't dropped, they're saved inside `this._initialClients` (see 1527) and then fed into the protocol handler when it's created.

Reply:
The `if` statement here is to ensure that in case of a reconnection, these are fed into the protocolhandler as soon as the reconnection happens.

Reply:
Got it! Maybe make a comment here about it - it's not obvious.